src/main/java/gov/nasa/jpl/mbee/mdk/generator/DocumentWriter.java

```diff
@@ -24,12 +41,16 @@
     private File realfile;
     private File dir;
     private boolean genNewImage;
-
+    private static FopFactory fopFactory;
+    
     public DocumentWriter(Document dge, File realfile, boolean genNewImage, File dir) {
```

Comment:
I don't think a user would often want to generate multiple formats. We should split this up into multiple MDActions, one that generates DocBook XML and another that generates a PDF. The latter will likely leverage all the code of the former and then convert. I think the menu path should be "DocGen" > "Generate..." > ["DocBook", "PDF"]. Each of those actions should force the appropriate file type (does not currently). Check out usages of `com.nomagic.magicdraw.actions.MDActionsCategory` in the project to see how actions can be nested. A bit of a scope creep I know, but it will help the user experience.

Additional resources: `gov.nasa.jpl.mbee.mdk.mms.actions.GenerateLocalDocument` (which should likely be renamed to `GenerateDocBookDocument`) and `gov.nasa.jpl.mbee.mdk.MDKConfigurator`:262